### ISSUE TITLE
[Feature] Zendesk Integration

### ISSUE THREAD

**Issue by felixsuarez0727** [appsmithorg/appsmith#13840]
Zendesk is a service-first CRM company that builds software designed to improve customer relationships. As employees, they encourage each other to grow and innovate. 

## Zendesk API Hierarchy

The API Structure is roughly divided like this:

#### Ticketing API [#13841](https://app.zenhub.com/workspaces/be-coders-pod-6167d73fab4b07001219a3d0/issues/appsmithorg/appsmith/13841)
- Tickets
- Users
- Organizations
- Groups
- Apps
- Targets
- Side Conversation 

#### Help Center API [#13842](https://app.zenhub.com/workspaces/be-coders-pod-6167d73fab4b07001219a3d0/issues/appsmithorg/appsmith/13842)
- Articles
- Article Comments
- Article Labels
- Categories 
- Search
- Sections
- External Content Records 
- External Content Sources
- External Content Types

#### Live Chat API [#13843](https://app.zenhub.com/workspaces/be-coders-pod-6167d73fab4b07001219a3d0/issues/appsmithorg/appsmith/13843)
- Chats
- Agents
- Visitors
- Departments
- Real Time Chat 

#### Voice API [#13844](https://app.zenhub.com/workspaces/be-coders-pod-6167d73fab4b07001219a3d0/issues/appsmithorg/appsmith/13844)
- Phone numbers 
- Digital lines 
- Availabilities
- Greetings
- Stats

#### Sales CRM API
`Pending analysis`

#### Custom Data API [#13845](https://app.zenhub.com/workspaces/be-coders-pod-6167d73fab4b07001219a3d0/issues/appsmithorg/appsmith/13845)
- Object Types
- Object Records
- Relationship Types
- Relationship Records 
- Custom Objects Events 
- Custom Objects Jobs
- Profiles
- Events 

Below is the status (as of 2022-05-14) of the endpoints of each section (at the moment the documentation uses v2 as point of reference).

## References (Useful Links)
[
Zendesk API reference documentation](https://developer.zendesk.com/api-reference)

**Comment by rohan-arthur** [comment]
@felixsuarez0727 please keep [this](https://www.notion.so/appsmith/Grouping-of-Integrations-fc6c634752284c5db1b1740aea70c598) in consideration before starting with zendesk

**Comment by rohan-arthur** [comment]
not a priority at present, closing this issue
